lib/puppet/type/cert_key_bundle.rb

```diff
@@ -0,0 +1,98 @@
+Puppet::Type.newtype(:cert_key_bundle) do
+  desc 'combines a certificate and private key into a single file'
+
+  ensurable
+
+  newparam(:path, :namevar => true) do
+    desc "Path to the file that will contain the certificate and private key bundle"
+    isrequired
+  end
+
+  newparam(:certificate) do
+    desc "Path to certificate file to include in bundle"
+    isrequired
+
+    def insync?(is)
+      File.read(self[:path]).include?(File.read(should))
```

Comment:
I will apply these and the other comments over here on the stand-alone PR -- https://github.com/theforeman/puppet-certs/pull/380

Reply:
Correct. Since this is the private key and certificate in a single file I figure this `insync?` check should just check that the certificate is present.